test/extended/router/metrics.go

```diff
@@ -83,10 +84,21 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 		username, password = string(statsSecret.Data["statsUsername"]), string(statsSecret.Data["statsPassword"])
 
-		// Extract a bearer token from Prometheus authorized to access
-		// the router metrics URL.
-		bearerToken, err = findMetricsBearerToken(oc)
+		token, err := oc.AdminKubeClient().
+			CoreV1().
+			ServiceAccounts("openshift-monitoring").
+			CreateToken(
+				context.TODO(),
+				"prometheus-k8s",
+				&authenticationv1.TokenRequest{
+					Spec: authenticationv1.TokenRequestSpec{
+						Audiences: []string{"https://kubernetes.default.svc"},
```

Comment:
@ibihim @tkashem @s-urbaniak @soltysh changes like this one break on all clusters that do not have `kubernetes.default.svc` for the KAS SA account issuer configured, eg when using AWS STS. 

I am fixing this particlar instance in https://github.com/openshift/origin/pull/27120 but I would greatly appreciate it if you are mindful of this when reviewing future PRs.